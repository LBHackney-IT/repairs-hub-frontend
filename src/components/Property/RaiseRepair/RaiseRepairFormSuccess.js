import PropTypes from 'prop-types'
import Link from 'next/link'

const RaiseRepairFormSuccess = ({
  propertyReference,
  workOrderReference,
  shortAddress,
}) => {
  return (
    <div>
      <div className="govuk-panel govuk-panel--confirmation background-dark-green">
        <h1 className="lbh-heading-xl text-white">Repair work order created</h1>
        <div className="govuk-panel__body">
          Work order number
          <br></br>
          <strong className="govuk-!-font-size-48">{workOrderReference}</strong>
        </div>
      </div>

      <ul className="govuk-list govuk-!-margin-top-9">
        <li>
          <Link href={`/work-orders/${workOrderReference}`}>
            <a>
              <strong>View work order</strong>
            </a>
          </Link>
        </li>

        <li>
          <Link href={`/properties/${propertyReference}`}>
            <a>
              <strong>Back to {shortAddress}</strong>
            </a>
          </Link>
        </li>

        <li>
          <Link href="/">
            <a>
              <strong>Start a new search</strong>
            </a>
          </Link>
        </li>
      </ul>
    </div>
  )
}

RaiseRepairFormSuccess.propTypes = {
  propertyReference: PropTypes.string.isRequired,
  workOrderReference: PropTypes.number.isRequired,
  shortAddress: PropTypes.string.isRequired,
}

export default RaiseRepairFormSuccess
