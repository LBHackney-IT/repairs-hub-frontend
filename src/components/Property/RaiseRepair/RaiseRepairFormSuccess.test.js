import { render } from '@testing-library/react'
import RaiseRepairFormSuccess from './RaiseRepairFormSuccess'

describe('RaiseRepairFormSuccess component', () => {
  const props = {
    property: {
      propertyReference: '00012345',
      address: {
        shortAddress: '16 Pitcairn House  St Thomass Square',
      },
    },
    workOrderReference: 10000000,
  }

  it('should render properly', () => {
    const { asFragment } = render(
      <RaiseRepairFormSuccess
        propertyReference={props.property.propertyReference}
        shortAddress={props.property.address.shortAddress}
        workOrderReference={props.workOrderReference}
      />
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
