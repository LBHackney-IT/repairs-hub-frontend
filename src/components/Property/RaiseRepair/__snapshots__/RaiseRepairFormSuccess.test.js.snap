// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RaiseRepairFormSuccess component should render properly 1`] = `
<DocumentFragment>
  <div>
    <div
      class="govuk-panel govuk-panel--confirmation background-dark-green"
    >
      <h1
        class="lbh-heading-xl text-white"
      >
        Repair work order created
      </h1>
      <div
        class="govuk-panel__body"
      >
        Work order number
        <br />
        <strong
          class="govuk-!-font-size-48"
        >
          10000000
        </strong>
      </div>
    </div>
    <ul
      class="govuk-list govuk-!-margin-top-9"
    >
      <li>
        <a
          href="/work-orders/10000000"
        >
          <strong>
            View work order
          </strong>
        </a>
      </li>
      <li>
        <a
          href="/properties/00012345"
        >
          <strong>
            Back to 16 Pitcairn House  St Thomass Square
          </strong>
        </a>
      </li>
      <li>
        <a
          href="/"
        >
          <strong>
            Start a new search
          </strong>
        </a>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
